src/instructlab/model/backends/backends.py

```diff
@@ -117,29 +158,29 @@ def determine_backend(model_path: pathlib.Path) -> Tuple[str, str]:
     """
     Determine the backend to use based on the model file properties.
     Args:
-        model_path (Path): The path to the model file.
+        model_path (Path): The path to the model file/directory.
     Returns:
         Tuple[str, str]: A tuple containing two strings:
                         - The backend to use.
                         - The reason why the backend was selected.
     """
-
-    if model_path.is_dir():
-        if sys.platform != "linux":
-            raise ValueError(
-                "Model is a directory containing huggingface safetensors files but the system is not Linux. "
-                "Using a directory with safetensors file will activate the vLLM serving backend, vLLM is only supported on Linux. "
-                "If you want to run the model on a different system (e.g. macOS), please use a GGUF file."
-            )
-        # If the model is a directory, it's a VLLM model - it's kinda weak, but it's a start
+    
+    if model_path.is_dir() and is_model_safetensors(model_path):
         if sys.platform == "linux":
             logger.debug(
-                f"Model is a directory and system is Linux, using {VLLM} backend."
+                f"Model is a huggingface safetensors model and system is Linux, using {VLLM} backend."
             )
             return (
                 VLLM,
                 "model path is a directory containing huggingface safetensors files and running on Linux.",
             )
+        else:
+            raise ValueError(
```

Comment:
This is probably the `except` I mentioned above.

Reply:
done

Reply:
on second thought, i dont think this should be an exception, this should still be an else block as we want it to get invoked when nothing has gone wrong, we have just detected that we are running on non-linux and we should call that out

I will add a separate exception handler as well